workshops/chat/README.md

```diff
@@ -0,0 +1,311 @@
+# Chat
+
+Short link to this workshop: [https://workshops.hackclub.com/chat](https://workshops.hackclub.com/chat)
+
+Demo: [here](https://prophetorpheus.github.io/chat/)
+
+---
+
+In this workshop we'll be implementing a real-time chat using Firebase. Firebase is a back-end having many features, one of which is a real-time database. This means that all the information within the database gets updated across everything that uses the database.
+
+This workshop is a revival of the now-defunct tutorial made by Firebase itself.
+
+**Table of Contents**
+
+- [Part I: Set-up](#part-i-set-up)
+- [Part II: The HTML File](#part-ii-the-html-file)
+- [Part III: The JS File](#part-iii-the-js-file)
+- [Part IV: Publishing and Sharing and Chatting!](#part-iv-publishing-and-sharing-and-chatting)
+- [Part V: Hacking](#part-v-hacking)
+
+## Part I: Set-up
+
+- Open up the Cloud9 workspace
+- Make a new folder and name it `chat`
+- Make a new file called `index.html` in `chat`
+- Make another new file within `chat` and name it `main.js`
+- Type the base template into `index.html`. Here's what our `index.html` should look like after doing this:
+
+  ```js
+  <!DOCTYPE html>
+  <html>
+    <head>
+      <title>Chat with Me!</title>
+    <head>
+    <body>
+    </body>
+  </html>
+  ```
+- Include Firebase, by adding this to the **body** of `index.html`:
+
+  ```html
+  <script src="https://www.gstatic.com/firebasejs/live/3.0/firebase.js"></script>
+  ```
+
+- Now let's also add jQuery. Type the following script tag below the one that adds Firebase:
+
+  ```html
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
+  ```
+
+- Lastly, include the `main.js`, by adding this after the previous line:
+
+  ```html
+  <script src="main.js"></script>
+  ```
+
+## Part II: The HTML File
+
+Add a div to store message history in the **body**, above the script tags
+
+```html
+<div id="messages"></div>
+```
+
+Put these two inputs into the **body**, below the `div` we just created, and above the script tags:
+
+```html
+<input type="text" id="nameInput" placeholder="Name">
+<input type="text" id="messageInput" placeholder="Message">
+```
+
+Save and open up the external live preview.
+
+![](img/html_setup.png)
+
+## Part III: The JS File
+
+### Preparing to Use Firebase
```

Comment:
I think it may help to delve a bit into when Firebase can be used and why it is useful.